Realm/RLMProperty.mm

```diff
@@ -162,11 +160,38 @@ -(BOOL)setTypeFromRawType {
                     @throw RLMException([NSString stringWithFormat:@"'%@' is not supported as an RLMArray object type. RLMArrays can only contain instances of RLMObject subclasses. See http://realm.io/docs/cocoa/#to-many for more information.", self.objectClassName]);
                 }
             }
+            else if (strncmp(code, numberPrefix, numberPrefixLen) == 0) {
+                // get number type from type string - @"NSNumber<objectClassName>"
+                _type = RLMPropertyTypeArray;
+                _objectClassName = [[NSString alloc] initWithBytes:code + numberPrefixLen
```

Comment:
This is mimicking the array code in ways that doesn't make sense. Just check the suffix directly rather than creating an NSString and setting the ivars to nonsense values.